From 087ff37bb3c5a9ffdcaa0567f9ff39e13e4559d6 Mon Sep 17 00:00:00 2001
Message-Id: <087ff37bb3c5a9ffdcaa0567f9ff39e13e4559d6.1536075068.git.mprivozn@redhat.com>
In-Reply-To: <97f0c101bb0afba7a51eeb47e97c6e274f2ef16f.1536075068.git.mprivozn@redhat.com>
References: <97f0c101bb0afba7a51eeb47e97c6e274f2ef16f.1536075068.git.mprivozn@redhat.com>
From: Miguel Martin <mmartinv@redhat.com>
Date: Mon, 3 Sep 2018 14:44:38 +0200
Subject: [libvirt-snmp][PATCH] Fix wrong object OIDs sent by
 libvirtMib_subagent

https://bugzilla.redhat.com/show_bug.cgi?id=1624839

The objects OIDs sent by the guest agent are:
libvirtGuestName.0
libvirtGuestUUID.1
libvirtGuestState.2
libvirtGuestRowStatus.3

The expected libvirtGuestNotif objects OID would be:
libvirtGuestName.0
libvirtGuestUUID.0
libvirtGuestState.0
libvirtGuestRowStatus.0

Signed-off-by: Miguel Martin <mmartinv@redhat.com>
(cherry picked from commit 9dcfc08d8c967e66271d310d8f3f2630df751562)
Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
---
 src/libvirtNotifications.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/libvirtNotifications.c b/src/libvirtNotifications.c
index 135f177..0522d54 100644
--- a/src/libvirtNotifications.c
+++ b/src/libvirtNotifications.c
@@ -38,11 +38,11 @@ send_libvirtGuestNotif_trap(virDomainPtr dom)
     const oid libvirtGuestName_oid[] =
         { 1, 3, 6, 1, 4, 1, 36957, 1, 1, 1, 2, 0 };
     const oid libvirtGuestUUID_oid[] =
-        { 1, 3, 6, 1, 4, 1, 36957, 1, 1, 1, 1, 1 };
+        { 1, 3, 6, 1, 4, 1, 36957, 1, 1, 1, 1, 0 };
     const oid libvirtGuestState_oid[] =
-        { 1, 3, 6, 1, 4, 1, 36957, 1, 1, 1, 3, 2 };
+        { 1, 3, 6, 1, 4, 1, 36957, 1, 1, 1, 3, 0 };
     const oid libvirtGuestRowStatus_oid[] =
-        { 1, 3, 6, 1, 4, 1, 36957, 1, 1, 1, 9, 3 };
+        { 1, 3, 6, 1, 4, 1, 36957, 1, 1, 1, 9, 0 };
 
 
     const char *domName = virDomainGetName(dom);
-- 
2.16.4

